packages/node/src/policy-gen/policy-gen-context.js

```diff
@@ -26,29 +23,61 @@ import {
  *   ModuleSource,
  *   CompartmentSources,
  *   FileURLToPathFn} from '@endo/compartment-mapper'
+ * @import {VirtualModuleSource} from 'ses'
  * @import {Loggerr} from 'loggerr'
  * @import {LMRCache} from './lmr-cache.js'
- * @import {MissingModuleMap, PolicyGeneratorContextOptions} from '../internal.js'
+ * @import {CanonicalName, PolicyGeneratorContextOptions,
+ *   SimpleLavamoatModuleRecordOptions} from '../internal.js'
  * @import {LavamoatModuleRecord, IsBuiltinFn} from 'lavamoat-core'
  */
 
-const { entries, keys, fromEntries } = Object
+const { entries, keys, hasOwn } = Object
+
+/**
+ * Anything matching this is a specifier matching a package or a package and its
+ * subpath.
+ *
+ * Example _matching_ strings:
+ *
+ * - `foo`
+ * - `foo/bar`
+ * - `@foo/bar`
+ * - `@foo/bar/baz`
+ *
+ * Example _non-matching_ strings:
+ *
+ * - `./foo`
+ * - `./foo/bar`
+ * - `../foo/bar`
+ * - `../foo/bar/baz`
+ *
+ * The match is grouped into two groups; `pkg` for the package name and
+ * `subpath` for anything after (which may be `undefined`)
+ *
+ * @remarks
+ * Anything matching this is a specifier which should have an entry in the
+ * {@link LavamoatModuleRecord.importMap} record.
+ * @see {@link https://regexr.com/8crtf Regexr test}
+ */
+const PACKAGE_REGEX =
+  /^(?<pkg>(?:@[^/\s]+\/[^/\s]+)|[^@/\s.]+)(?:(?:\/)(?<subpath>(?:[^/\s/]+(?:\/[^/\s/]+)*)?))?$/
```

Comment:
- [ ] remove; check for leading `.`
- [ ] what happens if we use a relative path to import a different package?

Reply:
the answer is "it's reported as a missing module"